src/components/MarkersDisplay/Annotations/AnnotationLayerSelect.js

```diff
@@ -0,0 +1,155 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  parseExternalAnnotationPage,
+  parseExternalAnnotationResource
+} from '@Services/annotations-parser';
+
+const AnnotationLayerSelect = ({ annotationLayers = [], duration = 0, setDisplayedAnnotationLayers }) => {
+  const [selectedAnnotationLayers, setSelectedAnnotationLayers] = useState([]);
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedAll, setSelectedAll] = useState(false);
+
+  useEffect(() => {
+    if (annotationLayers?.length > 0) {
+      // Sort annotation sets alphabetically
+      annotationLayers.sort((a, b) => a.label.localeCompare(b.label));
+      // Select the first annotation set on page load
+      fetchAndParseLinkedAnnotations(annotationLayers[0]);
+    }
+  }, [annotationLayers]);
+
+  const isSelected = (layer) => selectedAnnotationLayers.includes(layer.label);
+  const toggleDropdown = () => setIsOpen((prev) => !prev);
+
+  /**
+   * Event handler for the check-box for each annotation layer in the dropdown
+   * @param {Object} annotationLayer checked/unchecked layer
+   */
+  const handleSelect = async (annotationLayer) => {
+    fetchAndParseLinkedAnnotations(annotationLayer);
+
+    // Uncheck and clear annotation layer in state
+    if (isSelected(annotationLayer)) clearSelection(annotationLayer);
+  };
+
+  /**
+   * Fetch linked annotations and parse its content only on first time selection
+   * of the annotation layer
+   * @param {Object} annotationLayer checked/unchecked layer
+   */
+  const fetchAndParseLinkedAnnotations = async (annotationLayer) => {
```

Comment:
What do you think about renaming this to represent the caching?  Rails has `find_or_create_*` methods for ActiveRecord models that will attempt to find first and create if not found.  This seems similar so maybe this could be `findOrFetchAndParseLinkedAnnotations` if it isn't too unwieldy?

Reply:
Yes, that sounds good!